onnxruntime/core/providers/trt/trt_allocator.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/allocator.h"
+
+namespace onnxruntime {
+constexpr const char* TRT = "Trt";
+
+class TRTPinnedAllocator : public CPUAllocator {
```

Comment:
>TRTPinnedAllocator [](start = 6, length = 18)

Some doxygen documentation for the two classes would be helpful as it's not obvious what OrtMemTypeCPU does differently to OrtMemTypeDefault.

Reply:
The default allocator doesn't allocate anything. The reason there are two TRT allocators is that due to an issue found by Cheng, a default allocator must be provided otherwise allocation planner won't get the default allocator information.

Reply:
This is great info to put in some comments here so it's easily found. 

---
In reply to: [261794030](https://github.com/Microsoft/onnxruntime/pull/382#discussion_r261794030) [](ancestors = 261794030)

Reply:
Added doxygen comments